### ISSUE TITLE
Support Enum types

### ISSUE THREAD

**Issue by RoyGoode** [realm/realm-dotnet#549]
#### Goal

Support Enum type properties in RealmObjects. Currently it is possible to persist the backing value by creating a separate property in the enum value type, such as byte or int, and ignoring the enum property. It would be more convenient for the developer if Enum type properties were supported by the Fody weaver.

#### Version of Realm and tooling
Realm version(s): 0.74.1

Xamarin/Visual Studio version: Xamarin Studio 6.0 (build 5156)

Which operating system version and device: Mac OS X 10.11.5

**Comment by AndyDentFree** [comment]
This is complicated a bit by having to reach agreement on what it means to other languages, so Realm files remain compatible.

See [Java request](https://github.com/realm/realm-java/issues/776) and [Swift request](https://github.com/realm/realm-cocoa/issues/921).

**Comment by dansiegel** [comment]
@AndyDentFree I might suggest that you look at how EntityFramework handles an enum by storing them as an int in the database and converting it back to the int when it initializes the object. That said it looks like you guys have this issue both here and #740. I might suggest bumping this up in priority.

**Comment by softlion** [comment]
Could you support enums plz ? It's the basic.  
It's simple integers ...

Or enable use of converters to bypass this weird limitation.

**Comment by papafe** [comment]
@softlion, you can actually already build a wrapper property around your enum, as explained in this [article](https://www.mongodb.com/developer/products/realm/advanced-modeling-realm-dotnet/#using-unpersistable-data-types)

**Comment by softlion** [comment]
> @softlion, you can actually already build a wrapper property around your enum, as explained in this [article](https://www.mongodb.com/developer/products/realm/advanced-modeling-realm-dotnet/#using-unpersistable-data-types)

That should be added somewhere to the doc.  
And also the generator could generate that code automatically :)

**Comment by papafe** [comment]
Yes, I agree that could be added to the docs to show some more advanced use cases. 
Regarding generating it... We could do it, but it starts to get a little complicated when you start changing the enums and then you need to convert them during migrations, for instance.

**Comment by softlion** [comment]
Or implement an attribute that activate and controls this conversion with an IValueConverter ?

**Comment by papafe** [comment]
This is another possibility, but we still would prefer to add enums as a supported data type, but that has his own challenges as you can see in the linked issues.

**Comment by softlion** [comment]
> This is another possibility, but we still would prefer to add enums as a supported data type, but that has his own challenges as you can see in the linked issues.

I prefer to let users decide if they want to store an enum as a string or an int.
